service/src/main/java/com/zebrunner/reporting/service/integration/tool/adapter/notificationservice/MicrosoftTeamsIntegrationAdapter.java

```diff
@@ -0,0 +1,225 @@
+
+public class MicrosoftTeamsIntegrationAdapter extends AbstractIntegrationAdapter implements NotificationServiceAdapter {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(MicrosoftTeamsIntegrationAdapter.class);
+
+    public static final String EMPTY_JSON_REQUEST_RESPONSE = "Summary or Text is required";
+
+    public static final Map<String, String> NOTIFICATION_COLOR = Map.of("PASSED", "22CA88", "FAILED", "D43A40", "SKIPPED", "FFD276");
+
+    private final String channelName;
+    private final String webHookValue;
+
+    private final UnirestInstance restClient;
+
+    public MicrosoftTeamsIntegrationAdapter(Integration integration) {
+        super(integration);
+
+        String webHook = getAttributeValue(integration, MicrosoftTeamsParam.MICROSOFT_TEAMS_WEB_HOOK);
+        Map<String, String> webhookData = getWebhookData(webHook);
+
+        this.channelName = webhookData.get("channelName");
+        this.webHookValue = webhookData.get("webHookValue");
+
+        this.restClient = initClient();
+    }
+
+    private UnirestInstance initClient() {
+        Config config = new Config();
+        config.addDefaultHeader("Accept", "application/json");
+        return new UnirestInstance(config);
+    }
+
+    private enum MicrosoftTeamsParam implements AdapterParam {
+        MICROSOFT_TEAMS_WEB_HOOK("MICROSOFT_TEAMS_WEB_HOOK");
+
+        private final String name;
+
+        MicrosoftTeamsParam(String name) {
+            this.name = name;
+        }
+
+        public String getName() {
+            return name;
+        }
+    }
+
+    @Override
+    public boolean isConnected() {
+        boolean connected = false;
+        try {
+            if (webHookValue != null) {
+                // Microsoft Teams webHook doesn't provide any functionality to check it's connectivity,
+                // so, to make sure it is valid and avoid test messages in channels, we can push empty object and validate response.
+                HttpResponse<JsonNode> response = restClient.post(webHookValue).body(new JsonObject()).asJson();
+                // Regardless of the webhook itself or its payload is incorrect, response status is always 400
+                if (response.getStatus() == 400) {
+                    // So we should perform an extra check and the only source of it is a response message
+                    UnirestParsingException responseMessage = response.getParsingError().orElse(null);
+                    if (responseMessage != null) {
+                        // We expect the message returned by API when valid webhook receives an empty Json payload we sent before,
+                        // the other way the connection considered not established
+                        connected = responseMessage.getOriginalBody().contains(EMPTY_JSON_REQUEST_RESPONSE);
+                    }
+                }
+            }
+        } catch (Exception e) {
+            LOGGER.error(e.getMessage());
```

Comment:
is it expected to log only error message?

Reply:
Since exception is constructed and adjusted with detailed message on the adapter level, we just need it here to be caught to prevent the other operations in stream from skipping when during initialization of one of the adapters was thrown exception. And it is logged as is in order not to miss that sth went wrong.